Snowflake.Data.Tests/UnitTests/Session/EasyLoggingStarterTest.cs

```diff
@@ -57,10 +74,84 @@ public void BeforeEach()
         {
             t_easyLoggingProvider = new Mock<EasyLoggingConfigProvider>();
             t_easyLoggerManager = new Mock<EasyLoggerManager>();
+            t_unixOperations = new Mock<UnixOperations>();
             t_directoryOperations = new Mock<DirectoryOperations>();
-            t_easyLoggerStarter = new EasyLoggingStarter(t_easyLoggingProvider.Object, t_easyLoggerManager.Object, t_directoryOperations.Object);
+            t_environmentOperations = new Mock<EnvironmentOperations>();
+            t_easyLoggerStarter = new EasyLoggingStarter(
+                t_easyLoggingProvider.Object,
+                t_easyLoggerManager.Object,
+                t_unixOperations.Object,
+                t_directoryOperations.Object,
+                t_environmentOperations.Object);
         }
-        
+
+        [Test]
+        public void TestThatThrowsErrorWhenLogPathAndHomeDirectoryIsNotSet()
+        {
+            // arrange
+            t_easyLoggingProvider
+                .Setup(provider => provider.ProvideConfig(ConfigPath))
+                .Returns(s_configWithNoLogPath);
+            t_environmentOperations
+                .Setup(env => env.GetFolderPath(Environment.SpecialFolder.UserProfile))
+                .Returns("");
+
+            // act
+            var thrown = Assert.Throws<Exception>(() => t_easyLoggerStarter.Init(ConfigPath));
+
+            // assert
+            Assert.IsNotNull(thrown);
+            Assert.AreEqual(thrown.Message, "No log path found for easy logging. Home directory is not configured and log path is not provided");
+        }
+
+        [Test]
+        public void TestThatThrowsErrorWhenLogPathIsNotSetAndHomeDirectoryThrowsAnException()
+        {
+            // arrange
+            var ex = new Exception("No home directory");
+            t_easyLoggingProvider
+                .Setup(provider => provider.ProvideConfig(ConfigPath))
+                .Returns(s_configWithNoLogPath);
+            t_environmentOperations
+                .Setup(env => env.GetFolderPath(Environment.SpecialFolder.UserProfile))
+                .Throws(() => ex);
+
+            // act
+            var thrown = Assert.Throws<Exception>(() => t_easyLoggerStarter.Init(ConfigPath));
+
+            // assert
+            Assert.IsNotNull(thrown);
+            Assert.AreEqual(thrown.Message, "No log path found for easy logging. Home directory is not configured and log path is not provided");
+        }
+
+        [Test]
+        public void TestThatDoesNotFailWhenDirectoryPermissionIsNot600Or700()
```

Comment:
I'm not sure what you mean.
We should fail if the config file is modifiable to others (we should check user and group permissions).
We should warn if there already was a directory for logs  `..../dotnet` but with different than 700 permissions.
Something is wrong here.

Reply:
This [test for EasyLoggingConfigFinder](https://github.com/snowflakedb/snowflake-connector-net/pull/849/files#diff-b439f18d56c386315b7fb79f6386b5f59ab62ed1d0890d01a688decb6ff8d35dR127) fails if the config file is modifiable by others

The test is for the logs directory but let me know if I misunderstood point 10 since currently it only fails for the client config file and only logs a warning for the logs directory

Reply:
Np, I'll revert the change and rename the test

Reply:
So maybe the last thing: let's add the information why it is dangerous to this warning: "Access permission for the logs directory is currently..." - we could maybe add that it could potentially be accessed by others. Then it would look more alarming.

Reply:
Ok

Reply:
The warning is updated